tldw_Server_API/app/core/AuthNZ/llm_budget_middleware.py

```
@@ -43,31 +74,112 @@
+                            f"SELECT id, user_id FROM api_keys "
+                            f"WHERE key_hash IN ({placeholders}) AND status = ? "
+                            f"ORDER BY created_at DESC LIMIT 1"
+                        )
+                        row = await pool.fetchone(query, (*digests, "active"))
+                        if row:
+                            key_id = row.get('id') if isinstance(row, dict) else row[0]
+                            try:
+                                request.state.api_key_id = key_id
+                                request.state.user_id = row.get('user_id') if isinstance(row, dict) else row[1]
+                            except Exception:
+                                pass
```

Ruff (0.14.2)

`try`-`except`-`pass` detected, consider logging the exception

(S110)